src/hooks/useDraftSpecEditor.ts

```diff
@@ -128,35 +131,16 @@ function useDraftSpecEditor(
         }
     }, [currentCatalog, draftSpecs, entityName, setSpecs]);
 
+    // This for advanced spec editor
     useEffect(() => {
-        if (currentCatalog && !isEqual(draftSpecRef.current, currentCatalog)) {
+        if (
+            monitorCurrentCatalog &&
+            currentCatalog &&
+            !isEqual(draftSpecRef.current, currentCatalog)
+        ) {
```

Comment:
Initially I removed this effect and the binding editor worked fine. However, this is how the Advanced Schema editor works. So added it back in with a flag. That way we are not constantly running updates that are just not needed.